### PR TITLE
KSP-121 Refactor add guests by email only

### PR DESCRIPTION
**Pull request by navarrojandg**
Current `functions:api` deployment breaks create event functionality on previous versions because a refactor on the `create-event` endpoint was done.

### CODE REVIEW

**Comment by navarrojandg** [comment]
After merging is complete we will need to make sure we deploy `functions:api` as well as update firestore rules.

File: functions/src/routes/createEvent.ts [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
 
 router.post(
     '/',
     createEventRecord,
+    fetchEventGuestsUID,
     createEventGuestsCol,
     updateEventGuestsRecord,
     updateEventHostsRecord,
-    emailEventGuests,
+    // emailEventGuests,
```
Comment (jesc0bar) [inline functions/src/routes/createEvent.ts:18]: I'll make sure to uncomment this before we merge.

File: firestore.rules [hunk 2]
```diff
@@ -21,18 +21,28 @@ service cloud.firestore {
...
       // check that the hostId does not change during the write
       return request.resource.data.hostId == resource.data.hostId;
     }
-    match /events/{document=**} {
-    	allow read: if isGuest() || isHost();
-      allow create: if true;
+    match /events/{EVENTID} {
+    	allow read: if isGuest(EVENTID);
+      allow create, delete: if isHost();
       allow update: if isHost() && isSafeEventWrite();
```
Comment (jesc0bar) [inline firestore.rules:35]: From these rules it seems like only the host can make changes to the event. Is that right?

**Review by jesc0bar — COMMENTED** [review]
Performed static testing and left a few comments. Need to carry out dynamic testing to conclude review and give approval.

File: firestore.rules (hunk 2 above)
Comment (navarrojandg) [reply firestore.rules:35]: Yeah, I'll need to add in another rule to allow updates for availability. Im considering moving availability to another subcollection similar to how it is for event guests. So we have finer control over read / write rules for event availability.

Comment (jesc0bar) [reply firestore.rules:35]: sounds like a good idea

**Review by jesc0bar — APPROVED** [review]

**Comment by navarrojandg** [comment]
Forcefully merging this pr after reviewing with Jorge. This pr is critical in fixing the functionality of adding guests to events and properly linking them.

**Pull request merged**